lib/p2p/Peer.ts

```diff
@@ -410,7 +437,7 @@ class Peer extends EventEmitter {
    * Waits for a packet to be received from peer.
    * @returns A promise that is resolved once the packet is received or rejects on timeout.
    */
-  private wait = (packetId: string, timeout?: number) => {
+  private wait = (packetId: string, timeout?: number): any => {
```

Comment:
I'd prefer avoiding `any`, I figure this can at least return `Promise<Packet>` and then when calling this method we can define the return type more specifically like:

```typescript
const inSessionAck: Packet<SessionAckPacketBody> = await this.wait(outSessionInit.header.id, Peer.RESPONSE_TIMEOUT);
```

Reply:
Good call, will change it. 

I'm afraid we're also not verifying that the response packet is of the expected type. This could lead to runtime errors. i'll open an issue for it.

Reply:
Right, we're only checking that the packet id matches what we expect. This crossed my mind as well and I agree it would be good to handle the case when the packet we receive doesn't have the type we expect.

Reply:
I opened #788